### ISSUE TITLE
C++17 build issue for MacOS

### ISSUE THREAD

**Issue by ajm-ska** [CARTAvis/carta-backend#502]
Just a reminder that somehow the C++ compiler on MacOS needs this set to 17 in order to accept the new C++17 functions used in this and the other hdf5-mipmaps branch.

So changing it to `set(CMAKE_CXX_STANDARD 17)` appears to fix the issue.

https://github.com/CARTAvis/carta-backend/blob/816f6ab6116d684f30eba638abdb3c41f57a3c29/CMakeLists.txt#L4

**Comment by veggiesaurus** [comment]
@ajm-asiaa do you want to make a PR for this? I see no reason why we shouldn't switch to C++17 if the compilers you're using to make the desktop versions support it.

**Comment by ajm-ska** [comment]
@veggiesaurus I did and then discovered it now fails on ubuntu... But works on CentOS7 (and MacOS)

https://carta.asiaa.sinica.edu.tw/blue/organizations/jenkins/carta-backend/detail/ajm%2Fc%2B%2B17/1/pipeline/10

**Comment by ajm-ska** [comment]
Our Ubuntu version is 16.04. At first I thought the problem was the gcc version (gcc 7.4.0) 
But it wasn't. 
Luckily I found the solution [here](https://github.com/everitoken/evt/issues/33) where they said C++17 support starts in cmake 3.8
Our Ubuntu 16.04 was using cmake `3.5.1`!
So I installed the latest cmake from source and now it works!

(I just need to update the ubuntu CircleCI Docker container too)
